### DISCUSSION TITLE
Improvements to locating `piccolo_conf.py`

### DISCUSSION THREAD

**Discussion by dantownsend** [piccolo-orm/piccolo#1371] [Ideas]
Issue https://github.com/piccolo-orm/piccolo/issues/323 gave me some ideas about improvements to the way Piccolo looks for a `piccolo_conf.py` file.

I'm just documenting them here as ideas - they could be developed in the future if people think they're needed.

## 1. `PICCOLO_CONF` changes

The `PICCOLO_CONF` environment variable could accept a comma separated list of paths, in case you wanted Piccolo to look in multiple places.

```bash
# Piccolo would search for piccolo_conf.py in the current directory, and in an 'app' sub directory
export PICCOLO_CONF="piccolo_conf,app.piccolo_conf"
```
## 2. Add `--piccolo_conf` option to the CLI

When using the `piccolo` CLI, an extra parameter could be added which specifies where to look for the `piccolo_conf.py` file:

```bash
piccolo migrations check --piccolo_conf=app.piccolo_conf
```

## 3. .piccolo file

The contents would be something like:

```bash
PICCOLO_CONF=piccolo_conf_test
```
The precedence for finding a `piccolo_conf.py` file would then be as follows:

 * If the `PICCOLO_CONF` environment variable exists, use this.
 * Otherwise, look for a `.piccolo` file in the current directory, and use that.
 * Otherwise, import `piccolo_conf.py`

**Comment by AliSayyah** [comment]
I like these ideas, especially `.piccolo`. it seems like a permanent solution.

**Comment by stronk7** [comment]
Just guessing if looking for it also in CWD could be interesting.

Until now, I’ve had it in the root of my project (following a hint in the docs) and it has been working perfectly from the IDE (because it, automatically, adds the project root to the python path.

But a few hours ago, I’ve packaged the project, that includes a script to launch it and when I install it in other environment/project and execute it, the conf file isn’t found anymore.

I know that I can I set it both adding to the python path or hacking the package main script to setup the piccolo_conf env var… and I will.  Just I was expecting it to work automatically in the root of any project / that uses to be the CWD normally.

FYC, ciao :)
